common/custom_claims.c

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/attestation/custom_claims.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/safemath.h>
+#include "oe_host_stdlib.h"
+
+static void _free_claim(oe_claim_t* claim)
+{
+    oe_free(claim->name);
+    oe_free(claim->value);
+}
+
+static oe_result_t _free_claims(oe_claim_t* claims, size_t claims_length)
+{
+    if (!claims)
+        return OE_OK;
+
+    for (size_t i = 0; i < claims_length; i++)
+        _free_claim(&claims[i]);
+    oe_free(claims);
+    return OE_OK;
+}
+
+static oe_result_t _add_claim(
+    oe_claim_t* claim,
+    const void* name,
+    size_t name_size, // Must cover the '\0' at end of string
+    const void* value,
+    size_t value_size)
+{
+    if (*((uint8_t*)name + name_size - 1) != '\0')
+        return OE_CONSTRAINT_FAILED;
+
+    claim->name = (char*)oe_malloc(name_size);
+    if (claim->name == NULL)
+        return OE_OUT_OF_MEMORY;
+    memcpy(claim->name, name, name_size);
+
+    claim->value = (uint8_t*)oe_malloc(value_size);
+    if (claim->value == NULL)
+    {
+        oe_free(claim->name);
+        claim->name = NULL;
+        return OE_OUT_OF_MEMORY;
+    }
+    memcpy(claim->value, value, value_size);
+    claim->value_size = value_size;
+
+    return OE_OK;
+}
+
+static oe_result_t _fill_with_custom_claims(
```

Comment:
This name confused me at first since I couldn't tell if it was meant to fill the buffer or fill the claims.
Only the "const" in 55 helped.  Suggest: _fill_array_with_custom_claims
```suggestion
static oe_result_t _fill_array_with_custom_claims(
```